pkg/scheduler/schedule_one.go

```diff
@@ -414,6 +414,10 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 		}
 	}
 	feasibleNodes, err := sched.findNodesThatPassFilters(ctx, fwk, state, pod, diagnosis, nodes)
+	// always try to update the sched.nextStartNodeIndex regardless of whether an error has occurred
+	// this is helpful to make sure that all the nodes have a chance to be searched
+	processedNodes := len(feasibleNodes) + len(diagnosis.NodeToStatusMap)
```

Comment:
The problem here is the `len(diagnosis.NodeToStatusMap)` is always zero, so we should 1) return the `processNodesNum` directly. 2) return the diagnosis. Currently, I prefer the second one.

Reply:
Hi @kerthcet 
`the len(diagnosis.NodeToStatusMap) is always zero` is NOT  really true.
You can have a try with this: https://go.dev/play/p/Gplw7S_XFZY

Reply:
My bad, I didn't notice the type.